src/udiskslinuxdriveata.c

```diff
@@ -1431,23 +1436,36 @@ handle_pm_standby (UDisksDriveAta        *_drive,
       goto out;
     }
 
-  {
-    /* ATA8: 7.55 STANDBY IMMEDIATE - E0h, Non-Data */
-    UDisksAtaCommandInput input = {.command = 0xe0};
-    UDisksAtaCommandOutput output = {0};
-    if (!udisks_ata_send_command_sync (fd,
-                                       -1,
-                                       UDISKS_ATA_COMMAND_PROTOCOL_NONE,
-                                       &input,
-                                       &output,
-                                       &error))
+  if (do_wakeup)
+   {
+      if (read (fd, buf, sizeof (buf)) != sizeof (buf))
```

Comment:
Wow, this has quite a lot of chances to fail. But I know you just moved the code. We may need to make it a bit more hardened in the future.